### ISSUE TITLE
Add FbProphet wrapper

### ISSUE THREAD

**Issue by robomotic** [alan-turing-institute/sktime#503]
Hi there,
it would be nice to add a wrapper for FbProphet (maybe as an optional dependency during pip install) that satisfies the SK interface in this way I will be able to compare it with other models in a uniform manner.

Cheers.

**Comment by mloning** [comment]
Hi @robomotic, yes, that'd be great! It's already on our forecasting Wishlist #220. 

Any help is much appreciated. 

There is also an interesting technical solution to handle these wrapper which one could try out, more details [here](https://gist.github.com/mloning/34aa0eb84cb14bedb9f878d446d51e5a).

**Comment by aiwalter** [comment]
I am working on this and will open a PR next days. The problem will be the dependency of `fbprophet` to `pystan` as the latter requires a C++ compiler. I am using Windows and already faced problems when installing `pystan`, so lets see how this will work :)

**Comment by robomotic** [comment]
That is great news, thanks for efforts and yes I know pystan is a challenge in terms of multiplatform deployments.

**Comment by mloning** [comment]
We usually handles this via soft dependencies, see for example the [`tsfresh`](https://github.com/alan-turing-institute/sktime/blob/master/sktime/transformers/panel/tsfresh.py) file. That way, sktime as a whole does not depend on it and only user who want to work with fbprophet have to install it.

**Comment by aiwalter** [comment]
@mloning great, thx for the hint. I will do it like that. 
Another question: `fbprophet` can also work with exogenous data (multivariate), should this also be considered? Afaik `sktime` provides currently only univariate time series regression algorithms?

**Comment by mloning** [comment]
Exogenous variables should work, some algorithms support it already e.g. `AutoARIMA`, the interface is also defined in the BaseForecaster

**Comment by aiwalter** [comment]
@mloning I have created a class `Prophet` with a `fit()` function. I am a bit stuck in writing the `predict()` function as I see you are using a couple of functions e.g. in `arima.py`. In this file, there is no docstring for some functions, so I was not 100% sure if I got everything.
Here my questions:

- what is `_predict_fixed_cutoff()` for?
- Is it mandatory to do a recursion in `_predict_in_sample()`, in `fbprophet` I can do it without. Is this ok?
- which functions are mandatory for the class `Prophet` to be either provided or inherited?

Thx, Martin

**Comment by mloning** [comment]
Hi @aiwalter, you need to inherit from class `[_SktimeForecaster`](https://github.com/alan-turing-institute/sktime/blob/06f14617596fb04ac6eebd248ffe1c1f1ceda72a/sktime/forecasting/base/_sktime.py#L25) and [`_OptionalForecastingHorizonMixin]`(https://github.com/alan-turing-institute/sktime/blob/06f14617596fb04ac6eebd248ffe1c1f1ceda72a/sktime/forecasting/base/_sktime.py#L417) and implement `_predict`. 

Does fbprophet provide a way to update the forecaster when new data is available? If so, you could also implement `update`.

**Comment by aiwalter** [comment]
@mloning Thank you, I am done with `_predict()` and `update()`. I have now just issues in installing via `python setup.py install` as I dont know how I have to set the environment variable `SKTIME_NO_OPENMP`. I actually just wanted to execute the tests and do `make docs`... Otherwise I can just open a PR and see what happens?

**Comment by mloning** [comment]
@aiwalter yes feel free to open the PR! 

Have you taken a look at our install instructions on our website? `pip install -e .[all_extras]` should work but yes depending on OS you may have to set a few additional flags. Which OS are you on?

**Comment by aiwalter** [comment]
@mloning I am on Windows 10, sry should have mentioned it. Yes, I was following this instructions..  So I got this message:
```
                    ***
It seems that sktime cannot be built with OpenMP support.

- If your compiler supports OpenMP but the build still fails, please
  submit a bug report at:
  'https://github.com/alan-turing-institute/sktime/issues'

- If you want to build sktime without OpenMP support, you can set
  the environment variable SKTIME_NO_OPENMP and rerun the build
  command. Note however that some estimators will run in sequential
  mode and their `n_jobs` parameter will have no effect anymore.

- See sktime advanced installation instructions for more info:
  'https://https://www.sktime.org/en/latest/installation.html'
                    ***
```

**Comment by mloning** [comment]
@aiwalter would you mind posting this on Gitter, someone might know how to fix this.

**Comment by mloning** [comment]
Closed by #515
